test/users/npadmana/blas/BLAS.chpl

```diff
@@ -0,0 +1,729 @@
+/*
+
+Compiling with BLAS
+-------------------
+
+In order to compile with the BLAS libraries, you will need
+BLAS and CBLAS (C wrappers to BLAS) installed on your system. You
+will also need to know the location of the cblas.h header file,
+as well as the BLAS libraries you need to link in (and where they
+are, if they're not in a standard location).
+
+We recommend using a version of BLAS optimized for your system
+to get the best performance.
+
+As an example, with OpenBLAS :
+
+.. code-block:: sh
+
+    chpl -I$PATH_TO_CBLAS_DIR \
+         -L$PATH_TO_BLAS_LIBS -lblas source.chpl
+
+You may need additional of different libraries. For instance, if you
+use ATLAS, you will likely need to also add a -latlas to the above
+command. We strongly recommend referring to the documentation for the
+version of BLAS you have.
+
+The CBLAS API
+-------------
+
+The C_BLAS submodule of the BLAS module provides access to all the 
+CBLAS calls. Arrays are passed in directly, while pointers to scalar
+quantities (including complex numbers) are passed by reference (removing
+the need to wrap these with ``c_ptrTo``). As with BLAS calls in C, 
+the user is responsible for passing in array dimensions etc. Furthermore, 
+different array element types require using different functions.
+
+We refer the user to the existing CBLAS documentation for the usage
+of these functions.
+
+Chapel Level 3 BLAS API
+-----------------------
+
+For convenience, we provide wrappers around the Level 3 (matrix-matrix)
+BLAS functions. These determine the appropriate functions to call, based
+on the array element types, as well as the array dimensions. The other
+functionality is identical to the corresponding BLAS functions.
+
+The names of these routines are identical to the corresponding BLAS functions,
+except that the type prefix is dropped. For instance, ``gemm`` is the 
+wrapper for the ``[sdcz]gemm`` routines.
+
+.. note::
+ Chapel determines the dimensions of the matrices from the arrays that are
+ passed in. However, if one is passing in a sub-array such that the array 
+ elements are not contiguously stored in memory, then the user needs to 
+ pass in the leading dimension (```lda``` etc) to the array, just as they
+ would in C. These default to 0, in which case Chapel determines the appropriate
+ values based on the array passed in.
+
+
+*/
+
+
+module BLAS {
+
+  use SysCTypes;
+  require "cblas.h";
+
+
+  /* Define row or column order */
+  enum Order {Row=101:c_int, Col=102};
+
+  /* Operation of matrix : none, transpose, or adjoint */
+  enum Op {N=111:c_int, T=112:c_int, H=113:c_int}; // NoTranspose, Transpose, Adjoint
```

Comment:
Yes, the first one was an anomaly -- I hadn't realized you could write it in this shortened way. Which I agree is nicer, so I've just updated things to be consistent.